authorize/evaluator/opa/policy/authz.rego

```diff
@@ -53,12 +53,67 @@ allow {
 # allow by impersonate domain
 allow {
 	some route
-	input.host = route_policies[route].source
+	allowed_route(input.url, route_policies[route])
 	allowed_user_domain(token.payload.impersonate_email)
 	token.valid
 	count(deny)==0
 }
 
+allowed_route(input_url, policy){
+	input_url_obj := parse_url(input_url)
+	allowed_route_source(input_url_obj, policy)
+	allowed_route_prefix(input_url_obj, policy)
+	allowed_route_path(input_url_obj, policy)
+	allowed_route_regex(input_url_obj, policy)
+}
+
+allowed_route_source(input_url_obj, policy) {
+	object.get(policy, "source", "") == ""
+}
+allowed_route_source(input_url_obj, policy) {
+	object.get(policy, "source", "") != ""
+	source_url_obj := parse_url(policy.source)
+	input_url_obj.host == source_url_obj.host
+}
+
+allowed_route_prefix(input_url_obj, policy) {
+	object.get(policy, "prefix", "") == ""
+}
+allowed_route_prefix(input_url_obj, policy) {
+	object.get(policy, "prefix", "") != ""
+	startswith(input_url_obj.path, policy.prefix)
+}
+
+allowed_route_path(input_url_obj, policy) {
```

Comment:
does this handle the trailing slash?

Reply:
I added a few more test cases to check. Seems to work as expected.